src/migrate/index.js

```diff
@@ -121,22 +138,89 @@ export default class Migrator {
     });
   }
 
+  _createMigrationLockTable(tableName) {
+    return this.knex.schema.createTable(tableName, function(t) {
+      t.integer('is_locked');
+    });
+  }
+
+  _getLockTableName() {
+    return this.config.tableName + '_lock';
+  }
+
+  _isLocked(trx) {
+    var tableName = this._getLockTableName();
+    return this.knex(tableName)
+      .transacting(trx)
+      .forUpdate()
+      .select('*')
+      .then(data => data[0].is_locked);
+  }
+
+  _lockMigrations(trx) {
+    var tableName = this._getLockTableName();
+    return this.knex(tableName)
+      .transacting(trx)
+      .update({ is_locked: 1 });
+  }
+
+  _getLock() {
+    return this.knex.transaction(trx => {
+      return this._isLocked(trx)
```

Comment:
I believe that is only the case if the isolation level is set to 'repeatable reads' or higher.

[Wikipedia](https://en.wikipedia.org/wiki/Isolation_(database_systems)#Read_committed) on 'read committed' isolation level or lower:
> ...read committed is an isolation level that guarantees that any data read is committed at the moment it is read. It simply restricts the reader from seeing any intermediate, uncommitted, 'dirty' read. It makes no promise whatsoever that if the transaction re-issues the read, it will find the same data; data is free to change after it is read.

So, if I understand correctly, at this isolation it is possible to have two transaction that issue a `select` before either transaction `update`s the lock table. This means they would each redundantly `update` the table and try to execute the migrations. Of course the chance of this happening would be greatly reduced since they would need to line up perfectly.

I'm not sure exactly how this would be configured since Knex doesn't have an isolation level interface - so perhaps this is a moot point.